### ISSUE TITLE
When beautify on save, don't unfold all the current code folds 

### ISSUE THREAD

**Issue by toddbluhm** [Glavin001/atom-beautify#116]
I have, beautify-on-save checked and when I save one of my js files, it unfolds all the code blocks I have folded up so then I have to go back and refold them all. This is by no means a high priority issue, but more something that would be a really nice enhancement to have.

I also understand this might be something completely out of atom-beautifiers control too.

**Comment by Glavin001** [comment]
Upon reflection I think this is something that should be handled inside of Atom. Unfortunately, I am not sure how they will accomplish this, however I suggest that their [`setText` function on the `TextEditor`](https://atom.io/docs/api/v0.196.0/TextEditor#instance-setText) somehow diffs the input from the current text state and knows how to properly setup the folds in relatively the same state as before.

I hope that this is something they will support in the future.

**Comment by Glavin001** [comment]
There is an implementation I found that looks promising.

Sublime-HTMLPrettify: https://github.com/victorporof/Sublime-HTMLPrettify
- Save: https://github.com/victorporof/Sublime-HTMLPrettify/blob/master/HTMLPrettify.py#L25-L28
- Restore: https://github.com/victorporof/Sublime-HTMLPrettify/blob/master/HTMLPrettify.py#L117-L125

**Comment by RiCoTeRoX** [comment]
something like that occurs with linter-jscs :(

**Comment by Szandor72** [comment]
Is there a little update on this? Constantly folding/unfolding does not make a lot of fun.

**Comment by teeramusic** [comment]
I would find this feature greatly useful as well.

**Comment by dlewand691** [comment]
Dear God, yes please. Have this feature would be a big time saver.

**Comment by Glavin001** [comment]
It looks like Atom now supports `setTextViaDiff`: https://atom.io/docs/api/v1.17.2/TextBuffer#instance-setTextViaDiff

> Replace the current buffer contents by applying a diff based on the given text.

This would be our best shot to get this to work -- let Atom deal with the changing of the text and state of the folds.

This would require changing https://github.com/Glavin001/atom-beautify/blob/8ead568182b16bf2aa0086568692ed0d88c823bf/src/beautify.coffee#L120 and testing it out.

Pull Requests very welcome!!

**Comment by Glavin001** [comment]
# Original (un-beautified)

```html
<!doctype html>
<html>

  <head>
    <title>Test Page</title>
  </head>

  <body>
    <h1>Hello</h1><p>
      World!
    </p>
  </body>

</html>
```

![image](https://cloud.githubusercontent.com/assets/1885333/26526216/317f5f7a-4349-11e7-948d-d60f3e95a806.png)

# Before (with folds)

![image](https://cloud.githubusercontent.com/assets/1885333/26526218/576936ca-4349-11e7-9597-2119cb7864f5.png)

# After

## Current Version (`setText`)

![image](https://cloud.githubusercontent.com/assets/1885333/26526221/68a56d6e-4349-11e7-8724-022270cb323b.png)

## Next Version (`setTextViaDiff`)

It worked! All folds are preserved.

![image](https://cloud.githubusercontent.com/assets/1885333/26526240/1ae250e6-434a-11e7-877b-67afe3e3db18.png)

And if I manually unfold:

![image](https://cloud.githubusercontent.com/assets/1885333/26526242/2eefa192-434a-11e7-8be2-49456e729c05.png)

Looks good!

**Comment by Glavin001** [comment]
Published to v0.29.26

**Comment by toddbluhm** [comment]
@Glavin001 Thanks so much! This is awesome! And thanks for not letting this issue die off.
